asserter/construction_test.go

```diff
@@ -299,55 +299,132 @@ func TestConstructionParseResponse(t *testing.T) {
 		},
 	}
 
-	for name, test := range tests {
-		asserter, err := NewClientWithResponses(
-			&types.NetworkIdentifier{
-				Blockchain: "hello",
-				Network:    "world",
-			},
-			&types.NetworkStatusResponse{
-				GenesisBlockIdentifier: &types.BlockIdentifier{
-					Index: 0,
-					Hash:  "block 0",
-				},
-				CurrentBlockIdentifier: &types.BlockIdentifier{
-					Index: 100,
-					Hash:  "block 100",
+	asserter, err := NewClientWithResponses(
+		&types.NetworkIdentifier{
+			Blockchain: "hello",
+			Network:    "world",
+		},
+		&types.NetworkStatusResponse{
+			GenesisBlockIdentifier: &types.BlockIdentifier{
+				Index: 0,
+				Hash:  "block 0",
+			},
+			CurrentBlockIdentifier: &types.BlockIdentifier{
+				Index: 100,
+				Hash:  "block 100",
+			},
+			CurrentBlockTimestamp: MinUnixEpoch + 1,
+			Peers: []*types.Peer{
+				{
+					PeerID: "peer 1",
 				},
-				CurrentBlockTimestamp: MinUnixEpoch + 1,
-				Peers: []*types.Peer{
+			},
+		},
+		&types.NetworkOptionsResponse{
+			Version: &types.Version{
+				RosettaVersion: "1.4.0",
+				NodeVersion:    "1.0",
+			},
+			Allow: &types.Allow{
+				OperationStatuses: []*types.OperationStatus{
+					{
+						Status:     "SUCCESS",
+						Successful: true,
+					},
 					{
-						PeerID: "peer 1",
+						Status:     "FAILURE",
+						Successful: false,
 					},
 				},
-			},
-			&types.NetworkOptionsResponse{
-				Version: &types.Version{
-					RosettaVersion: "1.4.0",
-					NodeVersion:    "1.0",
+				OperationTypes: []string{
+					"PAYMENT",
 				},
-				Allow: &types.Allow{
-					OperationStatuses: []*types.OperationStatus{
-						{
-							Status:     "SUCCESS",
-							Successful: true,
+			},
+		},
+	)
+	assert.NotNil(t, asserter)
+	assert.NoError(t, err)
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			err := asserter.ConstructionParseResponse(test.response, true)
+			if test.err != nil {
+				assert.Error(t, err)
+				assert.Contains(t, err.Error(), test.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+
+	var unsignedTest = map[string]struct {
```

Comment:
nit: you should add another value to the struct in `tests` (`signed bool`) instead of duplicating the entire testing loop here.